packages/terra-spacer/src/Spacer.jsx

```diff
@@ -4,108 +4,140 @@ import classNames from 'classnames/bind';
 import 'terra-base/lib/baseStyles';
 import styles from './Spacer.scss';
 
+import { mapShorthandToObject, shorthandValidator } from './_spacerShorthandUtils';
+import { SpacerSizes, SpacerClassMappings } from './_spacerConstants';
+
 const cx = classNames.bind(styles);
-const values = ['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'];
+
+/*
+  NOTE: this is needed in order to ensure that the types `marginTop`, `marginRight`, `paddingTop`, `paddingRight`, etc. are recognized by terra-props-table
+  as defined. Currently, simply putting `Object.values(SpacerSizes)` causes all other `marginX` and `paddingX` props to be recognized as `undefined`
+  in the PropsTable.
+*/
+const arrayOfSpacerSizes = [...Object.values(SpacerSizes)];
 
 const propTypes = {
   /**
    * Child Nodes.
    */
   children: PropTypes.node.isRequired,
+  /**
+   * Sets margin in a syntax similar to the standard CSS spec https://developer.mozilla.org/en-US/docs/Web/CSS/margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
```

Comment:
Lets redefine this prop to be something like:
`Sets the margin on all four sides of an element: margin-top, margin-right, margin-bottom, margin-left. This behavior is consistent to to the standard CSS spec https://developer.mozilla.org/en-US/docs/Web/CSS/margin

^ Same for padding description.